network.go

```diff
@@ -1057,16 +1059,15 @@ func (n *network) delete(force bool, rmLBEndpoint bool) error {
 	// We are about to delete the network. Leave the gossip
 	// cluster for the network to stop all incoming network
 	// specific gossip updates before cleaning up all the service
-	// bindings for the network. But cleanup service binding
-	// before deleting the network from the store since service
-	// bindings cleanup requires the network in the store.
+	// bindings for the network.
 	n.cancelDriverWatches()
 	if err = n.leaveCluster(); err != nil {
 		logrus.Errorf("Failed leaving network %s from the agent cluster: %v", n.Name(), err)
 	}
 
-	// Cleanup the service discovery for this network
-	c.cleanupServiceDiscovery(n.ID())
+	// Capture remaining binding info to clear after deleting
+	// the network from the store.
+	bindings = c.gatherBindingInfo(n.ID())
```

Comment:
fetching the info here, and unfortunately because the gatherBinding requires the network, does not seem to give any more guarantee on the cleanup. I wonder if at this point we should just leverage the cleaning done from the networkDB